homeassistant/components/mastodon/notify.py

```diff
@@ -60,8 +70,59 @@ def __init__(self, api: Mastodon) -> None:
         self._api = api
 
     def send_message(self, message: str = "", **kwargs: Any) -> None:
-        """Send a message to a user."""
+        """Toot a message, with media perhaps."""
+        data = kwargs.get(ATTR_DATA)
+
+        media = None
+        mediadata = None
+        target = None
+        sensitive = False
+        content_warning = None
+
+        if data:
+            media = data.get(ATTR_MEDIA)
+            if media:
+                if not self.hass.config.is_allowed_path(media):
+                    LOGGER.warning(f"{media} is not a whitelisted directory")
+                    return
+                mediadata = self.upload_media(media)
+
+            target = data.get(ATTR_TARGET)
+            sensitive = data.get(ATTR_MEDIA_WARNING)
+            content_warning = data.get(ATTR_CONTENT_WARNING)
+
+        if mediadata:
+            try:
+                self._api.status_post(
+                    message,
+                    media_ids=mediadata["id"],
+                    sensitive=sensitive,
+                    visibility=target,
+                    spoiler_text=content_warning,
+                )
+            except MastodonAPIError:
+                LOGGER.error("Unable to send message")
+        else:
+            try:
+                self._api.status_post(
+                    message, visibility=target, spoiler_text=content_warning
+                )
+            except MastodonAPIError:
+                LOGGER.error("Unable to send message")
+
+    def upload_media(self, media_path: Any = None) -> Any:
+        """Upload media."""
+        with open(media_path, "rb"):
+            media_type = self.media_type(media_path)
         try:
-            self._api.toot(message)
+            mediadata = self._api.media_post(media_path, mime_type=media_type)
         except MastodonAPIError:
-            LOGGER.error("Unable to send message")
+            LOGGER.error(f"Unable to upload image {media_path}")
+
+        return mediadata
+
+    def media_type(self, media_path: Any = None) -> Any:
```

Comment:
This too should be protected
```suggestion
    def _media_type(self, media_path: Any = None) -> Any:
```